pkg/feature/conditions.go

```diff
@@ -64,14 +64,16 @@ func WaitForPodsToBeReady(namespace string) Action {
 				return false, err
 			}
 
+			pods := podList.DeepCopy()
```

Comment:
do we need `DeepCopy` here? or `podList.Items = filterEvictedPods(podList.Items)` is enough

Reply:
bad habit of mine, you are right there's no need as it's already done from the List method https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/cache/internal/cache_reader.go#L172